src/Attribute.cpp

```diff
@@ -0,0 +1,65 @@
+#if HAVE_CONFIG_H
+#  include <config.h>
+#endif
+
+#ifdef WIN32
+#pragma warning(disable : 4786)   /* Disable annoying warning messages */
+#endif
+
+#include "random.h"
+#include "OutputMgr.h"
+#include "Attribute.h"
+
+void
+AttributeGenerator::GenerateAllAttributes(std::ostream &out)
+{
+	bool attr_emitted = false;
+	vector<Attribute*>::iterator itr;
+	for(itr = attributes.begin(); itr != attributes.end(); itr++){
+		string attr_generated;
+		attr_generated = (*itr)->OutputAttributes();
+		if(attr_generated != ""){
+			if(!attr_emitted){
+				out << " __attribute__((" << attr_generated;
+				attr_emitted = true;
+			}
+			else
+				out << ", " << attr_generated;
+		}
+	}
+	if(attr_emitted)
+		out << "))";
+}
+
+Attribute::Attribute(string name, int prob)
+	:  attribute(name), attribute_probability(prob)
+{
+}
+
+BooleanAttribute::BooleanAttribute(string name, int prob)
+	: Attribute(name, prob)
+{
+}
+
+string
+BooleanAttribute::OutputAttributes()
```

Comment:
Rename to "make_random" to be consistent with other language constructs.

Reply:
Sure. Done